WordPress/src/test/java/org/wordpress/android/viewmodel/posts/PostListItemUiStateHelperTest.kt

```diff
@@ -72,10 +82,117 @@ class PostListItemUiStateHelperTest {
+        assertThat(state.data.statusLabels).contains(UiStringRes(R.string.post_status_pending_review))
+    }
+
+    @Test
+    fun `local draft label shown for local posts`() {
+        val state = createPostListItemUiState(post = createPostModel(isLocalDraft = true))
+        assertThat(state.data.statusLabels).contains(UiStringRes(R.string.local_draft))
+    }
+
+    @Test
+    fun `locally changed label shown for locally changed posts`() {
+        val state = createPostListItemUiState(post = createPostModel(isLocallyChanged = true))
+        assertThat(state.data.statusLabels).contains(UiStringRes(R.string.local_changes))
+    }
+
+    @Test
+    fun `version conflict label shown for posts with version conflict`() {
+        val state = createPostListItemUiState(unhandledConflicts = true)
+        assertThat(state.data.statusLabels).contains(UiStringRes(R.string.local_post_is_conflicted))
+    }
+
+    @Test
+    fun `uploading post label shown when the post is being uploaded`() {
+        val state = createPostListItemUiState(uploadStatus = createUploadStatus(isUploading = true))
+        assertThat(state.data.statusLabels).contains(UiStringRes(R.string.post_uploading))
+    }
+
+    @Test
+    fun `uploading media label shown when the post's media is being uploaded`() {
+        val state = createPostListItemUiState(uploadStatus = createUploadStatus(hasInProgressMediaUpload = true))
+        assertThat(state.data.statusLabels).contains(UiStringRes(R.string.uploading_media))
+    }
+
+    @Test
+    fun `queued post label shown when the post has pending media uploads`() {
+        val state = createPostListItemUiState(uploadStatus = createUploadStatus(hasPendingMediaUpload = true))
+        assertThat(state.data.statusLabels).contains(UiStringRes(R.string.post_queued))
+    }
+
+    @Test
+    fun `queued post label shown when the post is queued for upload`() {
+        val state = createPostListItemUiState(uploadStatus = createUploadStatus(isQueued = true))
+        assertThat(state.data.statusLabels).contains(UiStringRes(R.string.post_queued))
+    }
+
+    @Test
+    fun `error uploading media label shown when the media upload fails`() {
+        val state = createPostListItemUiState(
+                uploadStatus = createUploadStatus(uploadError = UploadError(MediaError(AUTHORIZATION_REQUIRED)))
+        )
+        assertThat(state.data.statusLabels).contains(UiStringRes(R.string.error_media_recover_post))
+    }
+
+    @Test
+    fun `error uploading post label shown when the post upload fails`() {
+        val errorMsg = "testing error message"
+        val state = createPostListItemUiState(
+                uploadStatus = createUploadStatus(uploadError = UploadError(PostError(GENERIC_ERROR, errorMsg)))
+        )
+        assertThat(state.data.statusLabels).contains(UiStringText(errorMsg))
+    }
+
+    @Test
+    fun `error label has precedence over info labels`() {
```

Comment:
⚠️  I think _precedence_ may be the wrong word here. I was expecting the `statusLabels` to have `[error, info]` which is what _precedence_ is. Maybe this is better?

```kt
fun `given a mix of info and error statuses, only the error status is shown`() 
```

Or something along those lines.

Reply:
Ohh, my bad. I thought "precedence" can be used in context with both priority and order.

Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/9491/commits/7c2c422e2f83ff146ca1a12ab3593c18d2207c13